### PR TITLE
Issue/7579 featured image confirmation

### LINKED ISSUE #7579: Add confirmation after selecting featured image
When the user taps a featured image from the photo picker, the image is immediately uploaded and set as the featured image. It should instead behave like the new stock photos feature, which selects the image and shows a button to confirm the selection.

Note: #7574 should be tackled prior to this.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #7579 - This PR is a refactoring of the device photo picker, which previously treated tapping a photo as an immediate selection. For example, if the user was choosing a featured image, simply tapping a photo would immediately upload it and set it as the featured image. This was inconsistent with how selection works in the new stock photo picker, plus it was simply annoying because users shouldn't expect a simple tap to result in uploading, etc.

So, this PR adds a confirmation step (via the CAB) so the user has a chance to approve the selection before anything gets uploaded.

To test, open a post in the editor and set a featured image, or tap the gravator on the Me page to select a new gravatar image.

single photo picker

### CODE REVIEW

**Comment by hypest** [comment]
Did a quick test and it seems that the selection is lost on rotation @nbradbury .

Also, this is more weird, it seems that the list of thumbnails doesn't scroll fully in landscape on my Nexus 5X and the bottom images get overshadowed by the bottom bar.

**Comment by hypest** [comment]
Thanks for the single-selection mode treatment @nbradbury . An issue I noticed is that after selecting a thumbnail and canceling the mode (say, tap "Back"), the thumbnail still shows as selected.

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java [hunk 1]
```diff
@@ -213,49 +218,63 @@ private boolean isValidPosition(int position) {
...
+            }
+            mSelectedItems.add(position);
         } else {
-            mSelectedUris.add(item.mUri);
-            isSelected = true;
-            holder.mTxtSelectionCount.setText(String.format(Locale.getDefault(), "%d", mSelectedUris.size()));
+            int selectedIndex = mSelectedItems.indexOf(position);
+            if (selectedIndex > -1) {
+                mSelectedItems.remove(selectedIndex);
+            }
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java]: Can we simplify the search+removal into just `mSelectedItems.remove((Integer) position)` perhaps?

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java [hunk 2]
```diff
@@ -213,49 +218,63 @@ private boolean isValidPosition(int position) {
...
         }
-        holder.mTxtSelectionCount.setSelected(isSelected);
 
-        // animate the count
-        AniUtils.startAnimation(holder.mTxtSelectionCount, R.anim.pop);
+        ThumbnailViewHolder holder = getViewHolderAtPosition(position);
+        if (holder != null) {
+            if (canMultiselect()) {
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java]: The logic to update the `mTxtSelectionCount` here looks almost identical [to what happens in onBindViewHolder](https://github.com/wordpress-mobile/WordPress-Android/blob/0055e8fdf62265ec8e7d465daf98030cf191d96e/WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java#L168-L177). Can we perhaps unify it and extract it into a new method?

**Comment by nbradbury** [comment]
@hypest I believe I've addressed all the issues you raised above, ready for another round!

**Comment by nbradbury** [comment]
> It seems that on rotation the selected item is preserved, but the 'selected' mode is not: the "checkmark" CAB action is lost and the bottom bar is still visible.

Fixed in 312eb28.

> when a thumbnail gets deselected, its bubble icon animates OK but the icon itself is depicted as deselected

Fixed in 57b4397.

**Comment by nbradbury** [comment]
> On the topic of rotation, not sure if this should be in scope for this PR but, the (bottom sheet) media picker seems to lose its selected items when trying to add media to a post body and rotate the device.

I had noticed that while testing, too. Let's make that a separate issue, since it existed prior to this PR.

**Comment by hypest** [comment]
Thanks for the updates @nbradbury , :shipit: !

**Pull request merged**